### ISSUE TITLE
Some customers encountered the problem of not being able to mobilize their wallets to publish during the publishing process.

### ISSUE THREAD

**Issue by Erreth-Akbe** [SparkNFT/SparkLink#192]
## What's going wrong?

A user encountered a problem that the publishing operation could not continue.
![image](https://user-images.githubusercontent.com/43749085/180151330-7487ca7a-15b2-44be-b0b0-8ea919805333.png)
Console and Network erro record:
![0d42dbf20126f13247bb4ff46f06e97](https://user-images.githubusercontent.com/43749085/180152923-a44ae18a-03c9-45ad-962d-1fdcb5896fd1.png)
![image](https://user-images.githubusercontent.com/43749085/180152856-45b19266-cdfa-4aaa-ae1e-5b70a0805a47.png)
The user switched two PCs, one mobile phone, and tried wifi and mobile networks in the network environment, but they were all stuck at this step uniformly.
After the release and decryption is completed, the wallet is not activated, or the decryption is not completed.
The additional information is: When the user tries to encrypt a file of about 200mb, it will cause the webpage to crash, that is, the AES encryption process should be started normally.

reference may be useful:
https://stackoverflow.com/questions/46468104/how-to-use-subtlecrypto-in-chrome-window-crypto-subtle-is-undefined

## How to reproduce?

We can not reproduce this bug in our machine.
Or we just find the cause of this bug and adjust the user's environment to suit our product.

## Expected behavior

The publish process should work smoothly.

## Context

refactor branch.

**Comment by SantanaSP** [comment]
Do you believe we got an alien who had a good sight just now? He can even figure out what the content in your blurred "Network" screenshot is!

**Comment by Erreth-Akbe** [comment]
> Do you believe we got an alien who had a good sight just now? He can even figure out what the content in your blurred "Network" screenshot is!

The content in the network is not very important, because the content in it does not change before and after the error is reported. I wrote in the statement above.

**Comment by Erreth-Akbe** [comment]
> Do you believe we got an alien who had a good sight just now? He can even figure out what the content in your blurred "Network" screenshot is!

I'll replace the screenshots of this, probably in 1-2 hours.

**Comment by SantanaSP** [comment]
I think it's almost like a network error or a hard problem to solve since we can't reproduce its behavior in our own environment. Do you record the network activities from before the site is loading? And have you noticed anything strange been returned by the server?

**Comment by Erreth-Akbe** [comment]
> I think it's almost like a network error or a hard problem to solve since we can't reproduce its behavior in our own environment. Do you record the network activities from before the site is loading? And have you noticed anything strange been returned by the server?

I don't quite agree that this is a network bug. If you are more sure, I will go to him to take a screenshot to see if there is any problem in the loading. You can follow the console error report to see what kind of environment this problem usually occurs in. This is likely to be a problem caused by a supply chain that is not supported in a certain environment.

**Comment by Erreth-Akbe** [comment]
![4c3db1c2a26f3ada73df3bbc5ed6fc3](https://user-images.githubusercontent.com/43749085/180219075-d634018f-7630-4a8a-ae63-7b31b66b7f9c.png)

**Comment by Erreth-Akbe** [comment]
![539fe978947635e7d0ea150e5940dde](https://user-images.githubusercontent.com/43749085/180219782-58b95f88-97cd-464e-8d04-03619b6daa8d.png)

**Comment by Erreth-Akbe** [comment]
We want to find a way to adjust this costomer's environment so that he can use our Dapp

**Comment by SantanaSP** [comment]
> 



> > I think it's almost like a network error or a hard problem to solve since we can't reproduce its behavior in our own environment. Do you record the network activities from before the site is loading? And have you noticed anything strange been returned by the server?
> 
> I don't quite agree that this is a network bug. If you are more sure, I will go to him to take a screenshot to see if there is any problem in the loading. You can follow the console error report to see what kind of environment this problem usually occurs in. This is likely to be a problem caused by a supply chain that is not supported in a certain environment.

But since we cannot reproduce it in our environment and follows the principle of "unacceptable as uncorroborated evidence", I think it's more rational to treat it as an occasional problem first. Just think that, if it's the fault of one of our dependencies, why 
do the other instances just function well?

**Comment by SantanaSP** [comment]
I noticed that there're no XHR requests in your Network screenshot, which means the activity between the client and the server has not been recorded. Under such an information insufficient circumstance I'm afraid I can't guess what's the true criminal, the only thing we can do is to get a new Network report or blame the supply chain endlessly.

**Comment by Erreth-Akbe** [comment]
I think this kind of supply chain library error should be a relatively common problem, I mean crypto.subtle this supply chain library. So according to the results of my current research, it is very likely that this user does not have https support, so what I want to know is how can I determine that his current environment is a non-HTTPs environment? Why are all three of his devices turned off https? How do I call him back to https? How do we enable https by default?


> I noticed that there're no XHR requests in your Network screenshot, which means the activity between the client and the server has not been recorded. Under such an information insufficient circumstance I'm afraid I can't guess what's the true criminal, the only thing we can do is to get a new Network report or blame the supply chain endlessly.

**Comment by SantanaSP** [comment]
The Rollup will automatically generate the vendor bundle using polyfills or other replacements to replace the built-in module of nodejs, including `https`. So I think it's most likely not an issue of `https`.

**Comment by SantanaSP** [comment]
I believe that the bottom layer network functionality package is using `browser` field in their packages to help the bundler to accomplish such work. Here is the specification for it: https://github.com/defunctzombie/package-browser-field-spec

**Comment by SantanaSP** [comment]
I insist on acquiring complete information about the context before we can dig deeper into the supply chain. If you noticed that even a request to the server hasn't happened, it's most likely what's going wrong with his chrome configurations, since the chrome profile sync mechanism could spread anything wrong to all the devices of a user.

**Comment by SantanaSP** [comment]
<del>Oh I see he's using Firefox, which version of Metamask is he using? Does he syncs any his data across devices?</del>

Have you tried to use WalletConnect to reproduce it?

**Comment by Erreth-Akbe** [comment]
> ~Oh I see he's using Firefox, which version of Metamask is he using? Does he syncs any his data across devices?~
> 
> Have you tried to use WalletConnect to reproduce it?

Yes, there was same error at mobile client . It is so strange.

**Comment by SantanaSP** [comment]
> > ~Oh I see he's using Firefox, which version of Metamask is he using? Does he syncs any his data across devices?~
> > Have you tried to use WalletConnect to reproduce it?
> 
> Yes, there was same error at mobile client . It is so strange.

Do you mean you can reproduce it with WalletConnect? If so, this issue can be taken into agenda.

**Comment by Erreth-Akbe** [comment]
> > > ~Oh I see he's using Firefox, which version of Metamask is he using? Does he syncs any his data across devices?~
> > > Have you tried to use WalletConnect to reproduce it?
> > 
> > 
> > Yes, there was same error at mobile client . It is so strange.
> 
> Do you mean you can reproduce it with WalletConnect? If so, this issue can be taken into agenda.

no, he can reproduce. It is weird.

**Comment by SantanaSP** [comment]
Yeah since I'm not familiar with WalletConnect actually there are no other things I can do to solve it. Of cause under the promise that you record the network activities correctly.
